src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -16,18 +35,23 @@ export const TodoFilter = () => (
         type="text"
         className="input"
         placeholder="Search..."
+        value={query}
+        onChange={event => handleChange(event.target.value)}
       />
       <span className="icon is-left">
         <i className="fas fa-magnifying-glass" />
       </span>
 
       <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
-        />
+        {query && (
+          <button
+            aria-label="delete"
+            data-cy="clearSearchButton"
+            type="button"
+            className="delete"
+            onClick={() => handleChange('')}
+          />
+        )}
```

Comment:
better to move wrapper condition to include span on 45 line
https://prnt.sc/dhJKNBdbIU2q